### PR TITLE
Music controller fixes

### PR DESCRIPTION
**Pull request by peppy**
- Fix filtering not processing layout after all items are filtered away.
- Make animation direction based on playlist order when needed.

### CODE REVIEW

**Comment by smoogipooo** [comment]
Before I merge this, can you fix up the nullref when any of the BeatmapMetadata.SearchableTerms are null?

Triggers on line 53 of SearchContainer (`return filterable.MatchingCurrentFilter = childTerms.Any(term => term.IndexOf(SearchTerm, StringComparison.InvariantCultureIgnoreCase) >= 0);`).

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**
